datafusion-cli/src/format/print_format.rs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Print format variants
+use arrow::csv::writer::WriterBuilder;
+use datafusion::arrow::record_batch::RecordBatch;
+use datafusion::arrow::util::pretty;
+use datafusion::error::Result;
+use std::io::{Read, Seek, SeekFrom};
+use tempfile::tempfile;
+
+/// Allow records to be printed in different formats
+#[derive(Debug, Clone)]
+pub enum PrintFormat {
+    Csv,
+    Aligned,
+}
+
+impl PrintFormat {
+    /// print the batches to stdout using the specified format
+    pub fn print_batches(&self, batches: &[RecordBatch]) -> Result<()> {
+        match self {
+            PrintFormat::Csv => {
+                // utilizing a temp file so that we can leverage the arrow csv writer
```

Comment:
I don't think you need to use an actual `tempfile` here. You can probably print directly to a `Vec<u8>` in memory. Here is an example of how to do so from IOx:

https://github.com/influxdata/influxdb_iox/blob/411cf134e9087198f74fe78ccb3c62e16344bf2f/influxdb_iox_client/src/format.rs#L133-L143

Reply:
thanks for the example!